Add class RefractivityTable. Allows fast calculation of the integrate…

fschlueter opened the pull request:
…d refractivity for straight lines in curved geometries using pre-calculated tables. Assumes a spherical earth.

The tables are anchored at the earths surface. For a given source and receiver the local zenith at the surface of the earth is calculated and along this line the refractivity is calculated. 

With the default value of 1000 zenith bins some features in the residual remain. With 10000 they vanish however the table takes quite long to initialize.

The comparison with the numerical calculation reveals some bias which is zenith angle dependent. However this is well below 0.1ns in light propagation time and thus not really import. Also within one event the introduced bias is very small.

Commit:
Add class RefractivityTable. Allows fast calculation of the integrated refractivity for straight lines in curved geometries using precalculated tables. Assums a spherical earth.

Commit:
No need for negative distances here. Still the distance grid could be improved.

cg-laser commented on radiotools/helper.py:
```diff
@@ -63,6 +65,93 @@ def datetime_to_UTC(dt):
+        return x0 + b * mult, y0 - a * mult, x0 - b * mult, y0 + a * mult
+
+
+def get_zenith_angle_at_earth(zenith, station_level):
```
function documentation is missing. Also in general, the input/output parameters should be defined with

```
Parameters
------------
variablea: float
    describtion of the variable

fschlueter replied:
I agree this is missing. I will add this during the next week.

cg-laser commented on radiotools/helper.py:
```diff
@@ -1,7 +1,9 @@
 #!/usr/bin/env python
 # -*- coding: utf-8 -*-
 from __future__ import absolute_import, division, print_function  # , unicode_literals
+from radiotools.atmosphere import models as atm
```
does this import trigger the calculation of the tables? If yes, please move it to the function where you need it.

fschlueter replied:
No it should not.

cg-laser reviewed (changes requested):
I didn't check if the code works but I have a few general comments. 

Can you add a documentation to all the functions/class? (I know I have been sloppy in the past, but for new function we should try to make a better job) . 

Do you want to add a unit test?

fschlueter commented:
Yes I can add a unit test.

Commit:
Add doc-strings to helper functions

Commit:
Update RefractivityTable class.

Fixed but in calculation of the table for the integrated refractivity in a flat atmosphere. Detail improvements: Adding linear interpolation when reaching the upper edge of several tables. Defined a max height which defines the max distance as function of the zenith angle. Add doc strings.

Commit:
Add unit tests for new RefractivityTable class

Commit:
Add new tests*.py file in travis CI config file

cg-laser commented on radiotools/atmosphere/refractivity.py:
```diff
@@ -6,80 +6,186 @@
     return 1 + a * np.exp(-b * h)
 
 
+def get_refractivity_between_to_points_numerical(p1, p2, atm_model, refractivity_at_sea_level, debug=False):
```
typo in function name "to" -> "two"

cg-laser reviewed (approved):
looks good except for the typo in the function name.

Commit:
Fix typo in function name

Commit:
Fix typo in function name here too.
